### PR TITLE
[USH-581] Follow Up - Endpoints to manually update Profile Details on mobile app

### PR DESCRIPTION
**Pull request by MMwandigha**
Updated the frontend code for image upload/delete feature
Missing is the backend implementation

### CODE REVIEW

**Comment by Ifycode** [comment]
Reviewing @MMwandigha

**Review by Ifycode — CHANGES_REQUESTED** [review]
Nice start @MMwandigha 
Continue by studying the other profile information that can save, see how they were implemented and possibly use the idea to save to the backend.

**Comment by MMwandigha** [comment]
@Ifycode Noted. Let me get into it

**Comment by MMwandigha** [comment]
@Ifycode You can now fully review this

File: apps/mobile-mzima-client/src/app/profile/information/components/profile-photo/profile-photo.component.ts [hunk 1]
```diff
@@ -1,21 +1,135 @@
...
+  }
+
+  changePhoto(file: File): void {
+    const reader = new FileReader();
+    console.log(reader);
+    console.log(file);
+    reader.onload = () => {
+      this.photo = reader.result as string;
+
+      this.mediaService.uploadFile(file, 'Test caption').subscribe((response: any) => {
```
Comment (ushahidlee) [inline apps/mobile-mzima-client/src/app/profile/information/components/profile-photo/profile-photo.component.ts]: Should probably change 'Test Caption'!

**Review by ushahidlee — CHANGES_REQUESTED** [review]
I was able to change my profile picture once and then I couldnt change it after that. 

Also, the behaviour is a bit unclear.: after I click "Set new photo" and choose the new picture, it updates in the UI but the "save changes" button is inactive.

I think the issues here are related to events.

File: apps/mobile-mzima-client/src/app/profile/information/components/profile-photo/profile-photo.component.ts (hunk 1 above)
Comment (MMwandigha) [reply apps/mobile-mzima-client/src/app/profile/information/components/profile-photo/profile-photo.component.ts]: Got it. Should we name it "Profile Picture" instead?

**Comment by MMwandigha** [comment]
@ushahidlee Hmm, that's strange.
Oh yeah, I hadn't taken note of the save button. Let me see what I can come up with.

File: apps/mobile-mzima-client/src/app/profile/information/components/profile-photo/profile-photo.component.ts (hunk 1 above)
Comment (ushahidlee) [reply apps/mobile-mzima-client/src/app/profile/information/components/profile-photo/profile-photo.component.ts]: Maybe the user's display name?

Comment (MMwandigha) [reply apps/mobile-mzima-client/src/app/profile/information/components/profile-photo/profile-photo.component.ts]: Yeah, I think this may work

**Comment by MMwandigha** [comment]
So here is my attempt at this ticket trying to achieve the following objectives (as shown by the design)

-Implementing the spinner to display once a profile picture is uploading
-Displaying the delete button only in instances where the user has edited the photo before. The delete button should be disabled as long as the default gravatar is shown
-Showing a toast message upon successful deletion

I pushed this code for extensive review before clean up by @ushahidlee , @Ifycode or @Angamanga . (We can arrange to meet). If you have any suggested changes, I can do them before pushing this code again

**Review by ushahidlee — CHANGES_REQUESTED** [review]
Code looks good and the functionality works, but I do not see the spinner while the picture is uploading. Otherwise its great!

**Comment by MMwandigha** [comment]
Thanks @ushahidlee I see...can you try with an image with a darker colors, perhaps the spinner was blending in with white photos. Also I think I should consider changing contrast based on the image color.

**Comment by MMwandigha** [comment]
This PR shows my progress towards optimizing the profile picture upload ticket. Here's what happened

- The user's custom profile picture will only be displayed after the photo has been fully uploaded to the server
- Improved the spinner's visibility by adding a stroke to the spinner. The border property didn't quite work because the spinner changes to white in dark mode and black in light mode

Pending: The profile picture is taking long to upload (approx 30 sec.) if you choose large images. @Angamanga is there a way to mitigate this?

**Comment by MMwandigha** [comment]
To prevent concurrent uploads, I created an emitter that blocks backward navigation as long as the photo is still uploading and also a toast message to tell the users that the photo is still uploading. @ushahidlee Let me know if it works well.

**Review by ushahidlee — APPROVED** [review]
Code looks good and works well!

**Pull request merged**